website/pages/blog/setting-up-graphql-server-to-scale-with-graphql-codegen-server-preset.mdx

```diff
@@ -0,0 +1,166 @@
+│   ├── server.ts
+│   ├── codegen.yml
+```
+
+While this works at a small scale where everyone knows each others, it would be hard to know who owns what as their could be hundreds of datasources and resolvers. A simple solution is to assign files to owners using GitHub's CODEOWNERS or similar features. However, it has to be done at a file level because each files are in category folders i.e. `resolvers`, `datasources`, etc.
+
+It is tempting to split this structure up into folders, each managed by a team:
+
+```markdown
+├── src/
+│   ├── schema/
+│   │   ├── TeamA/ # CODEOWNERS set up to ping TeamA if anything changes in this folder
+│   │   │   ├── datasources/
+│   │   │   │   ├── UserDatasource.ts
+│   │   │   │   ├── AuthDatasource.ts
+│   │   │   ├── resolvers/
+│   │   │   │   ├── userResolvers.ts
+│   │   │   │   ├── authResolvers.ts
+│   │   │   ├── userSchema.graphql
+│   │   │   ├── authSchema.graphql
+│   │   ├── TeamB/ # CODEOWNERS set up to ping TeamB if anything changes in this folder
+│   │   │   ├── datasources/
+│   │   │   │   ├── BookDatasource.ts
+│   │   │   ├── resolvers/
+│   │   │   │   ├── bookResolvers.ts
+│   │   │   ├── bookSchema.graphql
+│   ├── server.ts
+│   ├── codegen.yml
+```
+
+This is already a big improvement as ownership is clearer, but there's still organisational problems. For example, what happens when Team A changes what they own? Or split up into smaller teams? Or simply change their name? All of these scenarios require some level of admin work: at least renaming of the folder or at most moving files around, both having no values to end user.
+
+On top of that, all datasources and resolvers must be put together into the GraphQL server. In this case, it's the `server.ts` file. Who owns this file and other maintenance of the server such as `codegen.yml` (we all use GraphQL Codegen right? 😉), package updates, security patches, etc.?
+
+### Solution: Split into Modules
+
+It is fairly common to see teams change name or mission. However, the thing that does not change very often (and in some cases, never!) is business domain. If we split our code based on business domain and assign teams accordingly, it becomes much more scalable.
+
+On the maintenance of the server, it is best to have a small group of dedicated maintainers. This could consist of one member from each team in the codebase and membership can rotate, or another team altogether. Having a dedicated group of maintainers help reduce noise and cognitive load to the rest of the teams.
+
+Using the same example before, we can identify 3 main domains: User, Auth and Book, each have CODEOWNERS set up to notify them of changes. The Maintainers own `server.ts` and other configs.
+
+```markdown
+├── src/
+│   ├── schema/
+│   │   ├── user/ # CODEOWNERS set up to ping TeamA if anything changes in this folder
+│   │   │   ├── datasources/
+│   │   │   │   ├── UserDatasource.ts
+│   │   │   ├── resolvers/
+│   │   │   │   ├── userResolvers.ts
```

Comment:
very preferential thing but why not call the file name. Given it will live in `user/datasources` why to pre/suffix the source name?
```suggestion
│   │   │   ├── datasources/
│   │   │   │   ├── index.ts
│   │   │   ├── resolvers/
│   │   │   │   ├── index.ts
```

Reply:
Good call! I simplified it a bit further by just using files for this example. It looks something like this now:
```
├── src/
│   ├── schema/
│   │   ├── user/                     
│   │   │   ├── datasources.ts
│   │   │   ├── resolvers.ts
│   │   │   ├── schema.graphql
│   │   ├── auth/                     
│   │   │   ├── datasources.ts
│   │   │   ├── resolvers.ts
│   │   │   ├── schema.graphql
│   │   ├── book/                      
│   │   │   ├── datasources.ts
│   │   │   ├── resolvers.ts
│   │   │   ├── schema.graphql
│   ├── server.ts                      
│   ├── codegen.yml               
```